### PR TITLE
add package selection command line argument

### PR DESCRIPTION
**Pull request by onny**
Add the posibility to fetching only specific packages

```
./nc4nix -apps news,mail,calender
```

This interface is similar to the one in `wp4nix`

### CODE REVIEW

**Comment by leonklingele** [comment]
Thanks @onny! I'd suggest updating this with

```patch
diff --git a/main.go b/main.go
index 32e37c2..f6d4687 100644
--- a/main.go
+++ b/main.go
@@ -3,8 +3,8 @@ package main
 import (
 	"crypto/sha256"
 	"encoding/json"
-	"fmt"
 	"flag"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
@@ -159,7 +159,7 @@ func queryApi(v string) (ApiJson, error) {
 	return apiResponse, nil
 }
 
-func update(v string, hasLimits bool, limit string) {
+func update(v string, apps []string) {
 	log.Printf("Starting to process apps for version %s", v)
 	j, err := queryApi(v)
 	if err != nil {
@@ -184,18 +184,18 @@ func update(v string, hasLimits bool, limit string) {
 				na.Licenses = rel.Licenses
 			}
 		}
-		if hasLimits {
-			for _, pkgName := range strings.Split(limit, ",") {
-				if (a.Id == pkgName) {
+		if len(apps) > 0 {
+			for _, app := range apps {
+				if a.Id == app {
 					log.Printf("Found app %s (%s) at %s", a.Id, na.Version, na.Url)
 					an[a.Id] = na
+					break
 				}
 			}
 		} else {
 			log.Printf("Found app %s (%s) at %s", a.Id, na.Version, na.Url)
 			an[a.Id] = na
 		}
-
 	}
 
 	for k, _ := range an {
@@ -248,11 +248,10 @@ func main() {
 		}
 		NEXTCLOUD_VERSIONS = append(NEXTCLOUD_VERSIONS, v)
 	}
-	apps := flag.String("a", "", "apps to fetch - defaults to all")
+	apps := flag.String("apps", "", "Apps to fetch. Defaults to all")
 	flag.Parse()
-	hasLimits := *apps != ""
 
 	for _, v := range NEXTCLOUD_VERSIONS {
-		update(v.String(), hasLimits, *apps)
+		update(v.String(), strings.Split(*apps, ","))
 	}
 }

```

**Comment by onny** [comment]
Integrated the changes. While testing I noticed that this change breaks the old behaviour where nc4nix without the `-apps` parameter fetches all packages. With this change it fetches none when no `-apps` parameter is given

File: main.go [hunk 1]
```diff
@@ -237,8 +248,10 @@ func main() {
 		}
 		NEXTCLOUD_VERSIONS = append(NEXTCLOUD_VERSIONS, v)
 	}
+	apps := flag.String("apps", "", "Apps to fetch. Defaults to all")
+	flag.Parse()
 
 	for _, v := range NEXTCLOUD_VERSIONS {
-		update(v.String())
+		update(v.String(), strings.Split(*apps, ","))
```
Comment (leonklingele) [inline main.go:255]:
```suggestion
		if *apps == "" { // https://github.com/golang/go/issues/35130
			update(v.String(), nil)
		} else {
			update(v.String(), strings.Split(*apps, ","))	
		}
```

**Comment by leonklingele** [comment]
@onny sorry, didn't test my patch previously. Please see and test my suggestion above 😊

**Comment by onny** [comment]
Cool thanks, tested it and now it is working :)

**Pull request merged**